File: backend/transaction-service/transaction-domain/transaction-application-service/src/main/java/com/saver/system/transaction/service/domain/dto/create/CreateTransactionResponse.java
package com.saver.system.transaction.service.domain.dto.create;

import com.saver.system.domain.valueobject.Money;
import com.saver.system.transaction.service.domain.valueobject.TransactionAddress;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import javax.validation.constraints.NotNull;
import java.util.UUID;

@Getter
@Builder
@AllArgsConstructor
public class CreateTransactionResponse {
    @NotNull
    private final UUID userId;
    @NotNull
    private final UUID accountId;
    @NotNull
    private final Money transactionAmount;
    @NotNull
    private final TransactionAddress transactionAddress;
    @NotNull
    private final String message;
    @NotNull
    private final String transactionDataType;


}

File: backend/Services/transactions/src/domain/domain-application-service/dto/create/create-transaction-response.ts
import { Money, TransactionAddress, UUID } from "@machsan-tiggo/common";

export class CreateTransactionResponse {
  private userId: UUID;

  private accountId: UUID;

  private transactionAmount: Money;

  private message: string;

  private transactionAddress: TransactionAddress;

  setUserId(userId: UUID) {
    this.userId = userId;
    return this;
  }
  setAccountId(accountId: UUID) {
    this.accountId = accountId;
    return this;
  }
  setTransactionAmount(transactionAmount: Money) {
    this.transactionAmount = transactionAmount;
    return this;
  }
  setMessage(message: string) {
    this.message = message;
    return this;
  }

  setTransactionAddress(transactionAddress: TransactionAddress) {
    this.transactionAddress = transactionAddress;
    return this;
  }
}
